bc_obps/registration/api/user_operator.py

```diff
@@ -87,44 +87,74 @@ def save_operator(payload: UserOperatorOperatorIn, operator_instance: Operator,
         # create parent operator records
         operator_has_parent_operators: bool = payload.operator_has_parent_operators
         if operator_has_parent_operators:
-            po_operator_fields_mapping = {
-                "po_legal_name": "legal_name",
-                "po_trade_name": "trade_name",
-                "po_cra_business_number": "cra_business_number",
-                "po_bc_corporate_registry_number": "bc_corporate_registry_number",
-                "po_website": "website",
-            }
             for idx, po_operator in enumerate(payload.parent_operators_array):
-                new_po_operator_instance: ParentOperator = ParentOperator(
-                    child_operator=created_or_updated_operator_instance,
-                    operator_index=idx + 1,
+
+                existing_parent_operator_indices = list(
+                    ParentOperator.objects.filter(child_operator=created_or_updated_operator_instance).values_list(
+                        'operator_index', flat=True
+                    )
                 )
```

Comment:
I suggest modifying this code to enhance its readability and efficiency. Here is the updated code:
```python
existing_parent_operator_indices = list(created_or_updated_operator_instance.parent_operators.values_list('operator_index', flat=True))
```
We can move this code out of the loop and place it above `if operator_has_parent_operators:` statement. This way, we can avoid running the same query multiple times during the loop execution. Moreover, we can use `existing_parent_operator_indices` to check for my earlier comment. For instance, if the user removes all parent operators, `operator_has_parent_operators` would evaluate to false. However, if `existing_parent_operator_indices` is true, we must archive all existing parent operators related to this operator.

Reply:
Thanks for the refactor! This has to stay in the loop or else highest_index won't change and we just keep overwriting the same id.